samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -2217,15 +2217,13 @@ public double getCalcQScore(ItemGradingData data,  ItemDataIfc itemdata, Map<Int
 		  acceptableVariance = new BigDecimal(varianceString);
 	  }
 	  
-	  String userAnswerString = data.getAnswerText().replaceAll(",", "").trim();
+	  String userAnswerString = data.getAnswerText().replaceAll(",|[\\s]+", "");
```

Comment:
May we should have a regex that covers everything otherwise just going to be adding something else to it later!
So something like "[^\\d.-]" basically says filter out everything but digits, decimal, or dash?

Reply:
This seems like all it was doing was removing the commas, but this solution also didn't account for countries that used decimal as a separator instead of commas. This seems like it should be going through NumberFormat but that will still throw an error if there's spaces in the string but that would cause a potential loss of accuracy. I guess the locale issue is a separate one from this one though.

Reply:
Hi @ern @jonespm 

The initial code `data.getAnswerText().replaceAll(",", "").trim());` only filters out the first&latest blanks and commas (as they are thousands separator in U.S or U.K.) from the student's answer. As Brian Jones suggests we believe that the comma symbol problem should be addressed in other ticket like https://jira.sakaiproject.org/browse/SAK-36801.

Our PR only adds more accuracy in the blanks filter because it is more common that the student adds blanks in the beginning, end or inside the answer (e.g.:  + 5, - 7, 8. 6,  9 ). I believe that trying to filter out all characters that an student is not allowed to type could accidentally add errors.  The characters that an student is allowed to type are digits, dots, minus/plus symbols, the letter 'e' (scientific notation as 5e10) but may exist more allowed characters that I don't know. Taking into account that accidentally adding blanks is more common I believe the PR should not be changed. On the other hand, if you do not agree, we could change the regexp with [^e-\d.+]

Reply:
Hi @ern @jonespm 

Do you agree on merging this PR to trunk based on my last comment or do you prefer to change the regexp?

Reply:
Can you look at kernels NumberUtil and use that or add to it.

Reply:
Hi @ern 

I have tried to move the method to the NumberUtil class (kernel-util) but I am not able to do it: Samigo-services, which needs use "kernel-util", is deployed in the "api" but "kernel-util" is in the "components" so "Samigo-services" is not able to access to the "kernel-util" component. Taking this into account I suggest to keep the current PR. What do you think about?